addrmgr/addrmanager.go

```diff
@@ -745,30 +768,23 @@ func (a *AddrManager) HostToNetAddress(host string, port uint16, services wire.S
 			return nil, fmt.Errorf("no addresses found for %s", host)
 		}
 		ip = ips[0]
-	}
-
-	return wire.NewNetAddressIPPort(ip, port, services), nil
-}
 
-// ipString returns a string for the ip from the provided NetAddress. If the
-// ip is in the range used for Tor addresses then it will be transformed into
-// the relevant .onion address.
-func ipString(na *wire.NetAddress) string {
-	if IsOnionCatTor(na) {
-		// We know now that na.IP is long enough.
-		base32 := base32.StdEncoding.EncodeToString(na.IP[6:])
-		return strings.ToLower(base32) + ".onion"
+		na = wire.NetAddressV2FromBytes(time.Now(), services, ip, port)
+	} else {
+		// This is an non-nil IP address that was parsed in the else if
+		// above.
+		na = wire.NetAddressV2FromBytes(time.Now(), services, ip, port)
```

Comment:
If this branch is reached, then `ip` is either an IPv6 or IPv4 address. The parsing is a little funky because the above branch actually sets `ip` here. So `NetAddressV2FromBytes` should hit the ipv4/ipv6 case and return no problem